zproject/prod_settings_template.py

```diff
@@ -152,6 +152,21 @@
     # "zproject.backends.GenericOpenIdConnectBackend",  # Generic OIDC integration, setup below
 )
 
+## JWT-based API keys fetching.
+##
+## If you want to fetch API keys via JWT in Zulip, you need to set the
+## secret key and algorithm to use to validate JWT tokens received.
+JWT_FETCH_API_KEYS: Dict[str, Any] = {
+    # Subdomain Host(s) for which you want to enable JWT-based API keys fetching.
+    "zulip.example.com": {
```

Comment:
We should key these using the `string_id` / `subdomain`, not the full hostname -- it's more consistent with the codebase, and has some minor benefits when changing the domain of a Zulip installation.

Reply:
I can see where the consistency and benefits come from, nonetheless we discussed internally this matter during the development and agreed that a `string_id` / `subdomain` configuration would have been less friendly for people not understanding in details the concepts of Zulip realms and subdomains.
In particular when dealing with the root domain (e.g.`example.com`) a configuration accepting `subdomain` as a key would need `""` (empty string).
As far as we understand, the scenario of a single org on the room domain is the most frequent in the wild, hence the decision to use a new dictionary accepting subdomain hosts. In most cases, that subdomain host will also match the external host, leading to a very user-friendly setup.

We're available to discuss and change this of course, but should we decide to use a `subdomain` based conf, we suggest to revert to the already existing `JWT_AUTH_KEYS` dictionary, since the settings would be somehow duplicated.

Reply:
Yeah, I see the concern. The issue on our end is that we have very similar structures for OIDC and SAML that use the `string_id` / `subdomain`, so adding something that uses the domains instead would be significantly more complex for us to document than "It works the same as all the other per-subdomain authentication backend configuration". 

Can you start a thread on #backend about this design detail?

It's possible that we'll decide a reasonable change is to support either format to work for all authentication backends, for example.

Reply:
Started a new topic here.

https://chat.zulip.org/#narrow/stream/3-backend/topic/Replace.20subdomain.20ids.20with.20hosts.20in.20auth.20backend.20confs

Reply:
The PR has been updated, the code now re-uses the existing `JWT_ATH_KEYS` dict.